pubsub/pullstream.go

```diff
@@ -100,29 +103,33 @@ func (s *pullStream) get(spc *pb.Subscriber_StreamingPullClient) (*pb.Subscriber
 	if spc != s.spc {
 		return s.spc, nil
 	}
+	// we are about to open a new stream: if necessary, make sure the previous one is closed
+	if s.close != nil {
+		s.close()
```

Comment:
So if I'm reading this correctly, there's a desire to make sure the stream is closed, both here and in line 74 above. Why can't you call pullstream.cancel instead? Could you explain a bit more why an additional cancel func is necessary?

Reply:
Hi! So on L74 it's just to tidy up the context that's just been created rather than actually close the stream (since the stream creation would have errored out).

But an additional context and separate, corresponding cancel function are needed because the main `pullStream` ctx and cancel are meant to last for the whole lifecycle of `pullStream`, while the grpc streams that are leaked may have a shorter lifespan (precisely because of the retry, when you get back here on L115 after the initial stream creation), so need to be tied to a shorter-lived child context of the main `pullStream` one.

Reply:
Ack, yeah that makes sense. I'll approve and prioritize merging this week. Thanks for the contribution!

Reply:
Is there something I can help out with to get this into the next release? Suffering from the same memory leak as @itizir 🙏